galaxy_ng/app/tasks/publishing.py

```diff
@@ -1,12 +1,47 @@
 import logging
 
-from pulp_ansible.app.tasks.collections import import_collection
+from django.conf import settings
 from pulpcore.plugin.models import ContentArtifact
+from pulpcore.plugin.tasking import enqueue_with_reservation
+from pulp_ansible.app.models import AnsibleRepository
+from pulp_ansible.app.tasks.collections import import_collection
+
+from .promotion import add_content_to_repository, remove_content_from_repository
 
 log = logging.getLogger(__name__)
 
 VERSION_CERTIFIED = "certified"
 
+golden_name = settings.GALAXY_API_DEFAULT_DISTRIBUTION_BASE_PATH
+staging_name = settings.GALAXY_API_STAGING_DISTRIBUTION_BASE_PATH
+
+
+def import_and_move_to_staging(artifact_pk, **kwargs):
+    """Import collection version and move to staging repository.
+
+    Custom task to call pulp_ansible's import_collection() task then
+    enqueue two tasks to add to staging repo and remove from inbound repo.
+
+    This task will not wait for the enqueued tasks to finish.
+    """
+    inbound_repository_pk = kwargs.get('repository_pk')
+    import_collection(artifact_pk=artifact_pk, repository_pk=inbound_repository_pk)
```

Comment:
(more general thoughts)

Since it looks like we will be using https://github.com/rsinger86/django-lifecycle as part of the RBAC support, I'm wondering if that might be useful for this case in the future.  For example, when the created artifact hit's it's https://rsinger86.github.io/django-lifecycle/hooks_and_conditions/   "after_create" stage, then trigger the enqueue_with_reservation(add_content_to_repository, locks, args=task_args)  etc.

But again, that's just an idea for the future.